stdlib/string.ml

```diff
@@ -280,3 +280,98 @@ let get_int32_le s i = B.get_int32_le (bos s) i
 let get_int32_be s i = B.get_int32_be (bos s) i
 let get_int64_le s i = B.get_int64_le (bos s) i
 let get_int64_be s i = B.get_int64_be (bos s) i
+
+(* Spellchecking *)
+
+let uchar_utf_8_byte_decode_length = function
+  | '\x00' .. '\x7F' -> 1
+  | '\x80' .. '\xC1' -> 0
+  | '\xC2' .. '\xDF' -> 2
+  | '\xE0' .. '\xEF' -> 3
+  | '\xF0' .. '\xF4' -> 4
+  | _ -> 0
```

Comment:
Can we return `1` instead of `0` here and get rid of the `if` in `utf_8_char_length` below?

Reply:
Rather not, after this PR I want to expose this in `Uchar`. Returning `0` means that you have an UTF-8 decoding error what you do on a decoding error is application dependent.

Reply:
(That is if we return `1` we can no longer distinguish invalid UTF-8 starter bytes)

Reply:
OK!